server/src/test/java/org/elasticsearch/cluster/routing/allocation/decider/DiskThresholdDeciderUnitTests.java

```diff
@@ -471,16 +473,95 @@ public void testShardSizeAndRelocatingSize() {
-            assertEquals(10090L, sizeOfRelocatingShards(allocation, node, true, "/dev/null"));
+            assertEquals(10100L, sizeOfUnaccountedShards(allocation, node, false, "/dev/null"));
+            assertEquals(10090L, sizeOfUnaccountedShards(allocation, node, true, "/dev/null"));
         } else {
-            assertEquals(100L, sizeOfRelocatingShards(allocation, node, false, "/dev/null"));
-            assertEquals(90L, sizeOfRelocatingShards(allocation, node, true, "/dev/null"));
+            assertEquals(100L, sizeOfUnaccountedShards(allocation, node, false, "/dev/null"));
+            assertEquals(90L, sizeOfUnaccountedShards(allocation, node, true, "/dev/null"));
         }
     }
 
-    public long sizeOfRelocatingShards(RoutingAllocation allocation, RoutingNode node, boolean subtractShardsMovingAway, String dataPath) {
-        return DiskThresholdDecider.sizeOfRelocatingShards(
+    public void testTakesIntoAccountExpectedSizeForInitializingSearchableSnapshots() {
+        Metadata metadata = Metadata.builder()
+            .put(
+                IndexMetadata.builder("test")
+                    .settings(
+                        settings(Version.CURRENT).put("index.uuid", "1234")
+                            .put(INDEX_STORE_TYPE_SETTING.getKey(), SEARCHABLE_SNAPSHOT_STORE_TYPE)
+                    )
+                    .numberOfShards(3)
+                    .numberOfReplicas(1)
+            )
+            .build();
+
+        Index index = new Index("test", "1234");
+        String nodeId = "node1";
+        ShardRouting shard1 = ShardRoutingHelper.moveToStarted(
+            ShardRoutingHelper.initialize(
+                ShardRouting.newUnassigned(
+                    new ShardId(index, 1),
+                    false,
+                    PeerRecoverySource.INSTANCE,
+                    new UnassignedInfo(UnassignedInfo.Reason.INDEX_CREATED, "foo")
+                ),
+                nodeId,
+                10
+            ),
+            metadata
+        );
+        ShardRouting shard2 = ShardRoutingHelper.moveToStarted(
+            ShardRoutingHelper.initialize(
+                ShardRouting.newUnassigned(
+                    new ShardId(index, 2),
+                    false,
+                    PeerRecoverySource.INSTANCE,
+                    new UnassignedInfo(UnassignedInfo.Reason.INDEX_CREATED, "foo")
+                ),
+                nodeId,
+                20
+            ),
+            metadata
+        );
+        ShardRouting shard3 = ShardRoutingHelper.moveToStarted(
+            ShardRoutingHelper.initialize(
+                ShardRouting.newUnassigned(
+                    new ShardId(index, 3),
+                    false,
+                    PeerRecoverySource.INSTANCE,
+                    new UnassignedInfo(UnassignedInfo.Reason.INDEX_CREATED, "foo")
+                ),
+                nodeId,
```

Comment:
Can we add one additional shard that is added to another node? Just for checking that it does not impact it.